### PR TITLE
Remove margins around icon buttons

### LINKED ISSUE #2807: [BUG] Clickable icons have unnecessary margins
Thank you for submitting an issue. Please refer to our [issue policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)
for additional information about bug reports. For help with debugging your code, please refer to [Stack Overflow](https://stackoverflow.com/questions/tagged/mlflow).

**Please fill in this bug report template to ensure a timely and thorough response.**

### Willingness to contribute
The MLflow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix
for this bug to the MLflow code base?

- [x] Yes. I can contribute a fix for this bug independently.
- [ ] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.
- [ ] No. I cannot contribute a bug fix at this time.

### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **MLflow installed from (source or binary)**:
- **MLflow version (run ``mlflow --version``)**:
- **Python version**:
- **npm version, if running the dev UI**:
- **Exact command to reproduce**:

### Describe the problem
Describe the problem clearly here. Include descriptions of the expected behavior and the actual behavior.

This PR https://github.com/mlflow/mlflow/pull/2718 (more precisely this commit https://github.com/mlflow/mlflow/pull/2718/commits/9a8090ac06b815bba787909756019b75482b4208) replaced `<a>` tags used as a clickable icon with ant design's `<Button>` component. This replacement added unnecessary margins around the icons.

### Before:

Screen Shot 2020-05-10 at 23 49 59

### After:

Screen Shot 2020-05-10 at 23 49 30

### Code to reproduce issue
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

### Other info / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks,
please include the full traceback. Large logs and files should be attached.

### What component(s), interfaces, languages, and integrations does this bug affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [x] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

### LINKED ISSUE #2818: [BUG] Unable to delete experiments with long names from UI
Thank you for submitting an issue. Please refer to our [issue policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md) for additional information about bug reports. For help with debugging your code, please refer to [Stack Overflow](https://stackoverflow.com/questions/tagged/mlflow).

**Please fill in this bug report template to ensure a timely and thorough response.**

### Willingness to contribute
The MLflow Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the MLflow code base?

- [ ] Yes. I can contribute a fix for this bug independently.
- [ ] Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.
- [x] No. I cannot contribute a bug fix at this time.

### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**: no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: MacOS 10.15.4, Google Chrome
- **MLflow installed from (source or binary)**: Source
- **MLflow version (run ``mlflow --version``)**: 1.8.0
- **Python version**: 3.7
- **npm version, if running the dev UI**:
- **Exact command to reproduce**: mlflow ui

### Describe the problem
Experiments with long names in the sidebar are missing the delete icon. There is no way of deleting them from within the web interface.

mlflow-experiment-list-ui

### Code to reproduce issue
Create an experiment with a long name (using the UI or through any means).

### Other info / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

### What component(s), interfaces, languages, and integrations does this bug affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [x] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

Fixes #2807 and Fixes #2818 

Before:

Screen Shot 2020-05-15 at 0 37 45

After:

Screen Shot 2020-05-15 at 0 35 45

## How is this patch tested?

- Unit test
- Manually verified margins around icon buttons are removed.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [x] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
@Zangr Can you review this PR?

File: mlflow/server/js/src/common/components/IconButton.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Button } from 'antd';
+
+export class IconButton extends Component {
+  static propTypes = {
+    children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
+    style: PropTypes.object,
+    restProps: PropTypes.object,
```
Comment (Zangr) [inline mlflow/server/js/src/common/components/IconButton.js:9]: What do you think about adding an optional `type` prop, if present, automatically add children `<Icon type={props.type}` />`

File: mlflow/server/js/src/common/components/IconButton.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Button } from 'antd';
+
+export class IconButton extends Component {
```
Comment (Zangr) [inline mlflow/server/js/src/common/components/IconButton.js:5]: A functional component should be enough?

**Review by Zangr — APPROVED** [review]
@harupy Changes look great! I left some comments mostly around props support for us to discuss. Also could you please help check the icon alignment in other pages with the change?

File: mlflow/server/js/src/common/components/IconButton.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  static propTypes = {
+    children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
+    style: PropTypes.object,
```
[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/server/js/src/common/components/IconButton.js:8]: This makes sense. I'll add `className` to `props`.

File: mlflow/server/js/src/common/components/IconButton.js (hunk 1 above)
Comment (harupy) [reply mlflow/server/js/src/common/components/IconButton.js:9]: What about creating a `icon` prop that takes a JSX.Element?

```js
// antd icon

<IconButton
  ...
  icon={<Icon type="edit">}
/>

// fontawesome icon

<IconButton
  ...
  icon={<i className='far fa-trash-alt' />}
/>
```

File: mlflow/server/js/src/common/components/IconButton.js (hunk 2 above)
Comment (harupy) [reply mlflow/server/js/src/common/components/IconButton.js:5]: Yes. I'll fix.

File: mlflow/server/js/src/common/components/IconButton.js (hunk 1 above)
Comment (Zangr) [reply mlflow/server/js/src/common/components/IconButton.js:9]: LGTM!

**Comment by harupy** [comment]
Screen Shot 2020-05-21 at 21 52 10

---

Screen Shot 2020-05-21 at 21 52 38

---

Screen Shot 2020-05-21 at 21 52 25

**Comment by harupy** [comment]
@Zangr Can you review this PR again?

**Review by Zangr — APPROVED** [review]
@harupy Thank you for updating the PR! LGTM!

**Comment by msminhas93** [comment]
https://github.com/mlflow/mlflow/issues/2819 isn't fixed by this PR. The sidebar issue still persists. Could anyone of you investigate into it? Hoping for your support. Thanks!

**Pull request merged**